050_wpsage-plugin_L133.php

```
@@ -0,0 +1,389 @@
+    public function run_sql_query($request) {
+        global $wpdb;
+
+        $params = $request->get_json_params();
+        $query = isset($params['query']) ? $params['query'] : '';
+
+        if (empty($query)) {
+            return new WP_Error('invalid_query', 'SQL query is required', array('status' => 400));
+        }
+
+        // Whitelist allowed SQL operations for security
+        $allowed_operations = array('SELECT', 'SHOW', 'DESCRIBE', 'DESC');
```

PHPMD (2.15.0)

Avoid unused local variables such as '$allowed_operations'. (undefined)

(UnusedLocalVariable)